packages/code-combinators/test/Test/CodeCombinators/GenPat.hs

```diff
@@ -0,0 +1,106 @@
+module Test.CodeCombinators.GenPat where
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Language.Haskell.TH as TH
+import qualified Happy.Backend.CodeCombinators.Syntax as SnGen
+import qualified Happy.Backend.CodeCombinators.Abstract()
+import Happy.Backend.CodeCombinators
+import Test.CodeCombinators.Common
+import Data.List
+
+
+genLitP :: MonadGen m => m TH.Pat
+genLitP = do
+  x <- Gen.choice $
+         [
+            do
+            c <- Gen.latin1
+            return $ litP $ TH.CharL c
+          , do
+            s <- Gen.list (Range.linear 0 20) Gen.latin1
+            return $ litP $ TH.StringL $ delete '\n' s
```

Comment:
Hmm, deleting `\n` is a pretty suspicious thing to do? Why are you deleting specifically it? There are many other suspicious characters. Also, why specifically are you using `latin1` instead of full `unicode`?